Allow arbitrary environment names

NSeydoux opened the pull request:
We shouldn't use environment names as meaningful flags to enable or disable tests, but rather use specific feature flags. Thereforce, the environment name becomes only an informative property, and doesn't need any predefined values.

NSeydoux committed:
Allow arbitrary environment names

We shouldn't use environment names as meaningful flags to enable or
disable tests, but rather use specific feature flags. Thereforce, the
environment name becomes only an informative property, and doesn't need
any predefined values.

jeswr reviewed (commented):
If we are removing this gate, could you make any check that relies on environment variables vase insensitive. 

I find this is useful for helping me catch this like using a lower-case s for Podspaces in my configuration.

jeswr commented:
I would be in favour of a "custom" option.

NSeydoux committed:
Revert "Allow arbitrary environment names"

This reverts commit 2a65d2b53690e3b36418c98abc5b95854d071cff.

NSeydoux committed:
Change dev-next to dev-2-1 and add "Custom"

Change the supported environment name from "dev-next" to "dev-2-1", and
also add "Custom" when no fixed version correctly applies.
